src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -1132,6 +1125,33 @@ public async Task DisableTwoFactorProviderAsync(Organization organization, TwoFa
             }
         }
 
+        var organizationUsers = await SaveUsersSendInvitesAsync(organizationId, invites);
+
+        await _eventService.LogOrganizationUserEventsAsync(organizationUsers.Select(e => (e, EventType.OrganizationUser_Invited, (DateTime?)DateTime.UtcNow)));
+
+        return organizationUsers;
+    }
+
+    public async Task<List<OrganizationUser>> InviteUsersAsync(Guid organizationId, EventSystemUser systemUser,
+        IEnumerable<(OrganizationUserInvite invite, string externalId)> invites)
+    {
+        var organizationUsers = await SaveUsersSendInvitesAsync(organizationId, invites);
+
+        await _eventService.LogOrganizationUserEventsAsync(organizationUsers.Select(e => (e, EventType.OrganizationUser_Invited, systemUser, (DateTime?)DateTime.UtcNow)));
```

Comment:
You've moved the logging outside of the private method, but the private method still has `var events` which is collecting events as users are added.

Reply:
I've changed the private method to return the `events` list because I realised it may be different than the returned `organizationUsers` list.

Reply:
huh, the `orgUsers` returned by `InviteUsersAsync` only contains the admin-level users that were invited, not all the users invited. I doubt that callers of this method would expect that. But for now this looks good to me.